### ISSUE TITLE
Make Object use Redis Strucutres

### ISSUE THREAD

**Issue by pmvrmc** [shelf-js/shelf#12]
It would be great to have the object schema abstract redis structures, for example having an array in an object is internally really a list.

We could also use sets, maps, etc.

What do you think? This is a really broad and long term issue.

**Comment by ricardofbarros** [comment]
I'm with you on this @pmvrmc, the only problem is if your going to have performance losses instead of gains, because you need to parse the response on the js world. As JSON.parse is a cheap op (most of the times)

**Comment by JGAntunes** [comment]
Well, this will be a bit tricky, because you'll have to map a JS object to a couple of Redis data types and then establish relations between them (if I'm understanding this correctly). But I do agree that we should have a way for the programmer to say in the schema that he/she would like a specific key to be mapped into a specific data type.

**Comment by pmvrmc** [comment]
I agree this is tricky and might have performance losses, but it seems challenging and a lot of fun!
Shelf could act as an abstraction layer for a complex database structure :)

I will start thinking about this on the next couple of weeks.
